AIP 39: Documentation (#17552)
mistercrunch/Airflow: docs/apache-airflow/concepts/dags.rst

    with DAG("my_daily_dag", schedule_interval="0 * * * *"):
        ...

.. tip::

    For more information on ``schedule_interval`` values, see :doc:`DAG Run </dag-run>`.

    If ``schedule_interval`` is not enough to express the DAG's schedule, see :doc:`Timetables </howto/timetable>`.

> Every time you run a DAG, you are creating a new instance of that DAG which Airflow calls a :doc:`DAG Run </dag-run>`. DAG Runs can run in parallel for the same DAG, and each has a defined data interval, which identifies the *logical* date and time range it is running for - not the *actual* time when it was started.

Reviewer: ```suggestion
Every time you run a DAG, you are creating a new instance of that DAG which Airflow calls a :doc:`DAG Run </dag-run>`. DAG Runs can run in parallel for the same DAG, and each has a defined data interval, which identifies the period of data the tasks should operate on.
```

> Every time you run a DAG, you are creating a new instance of that DAG which Airflow calls a :doc:`DAG Run </dag-run>`. DAG Runs can run in parallel for the same DAG, and each has a defined data interval, which identifies the *logical* date and time range it is running for - not the *actual* time when it was started.

Reviewer: ```suggestion
Every time you run a DAG, you are creating a new instance of that DAG which Airflow calls a :doc:`DAG Run </dag-run>`. DAG Runs can run in parallel for the same DAG, and each has a defined data interval, which identifies the period of data the tasks should operate on.
```

    with DAG("my_daily_dag", schedule_interval="0 * * * *"):
        ...

.. tip::

    For more information on ``schedule_interval`` values, see :doc:`DAG Run </dag-run>`.

    If ``schedule_interval`` is not enough to express the DAG's schedule, see :doc:`Timetables </howto/timetable>`.

Every time you run a DAG, you are creating a new instance of that DAG which Airflow calls a :doc:`DAG Run </dag-run>`. DAG Runs can run in parallel for the same DAG, and each has a defined data interval, which identifies the *logical* date and time range it is running for - not the *actual* time when it was started.

As an example of why this is useful, consider writing a DAG that processes a daily set of experimental data. It's been rewritten, and you want to run it on the previous 3 months of data - no problem, since Airflow can *backfill* the DAG and run copies of it for every day in those previous 3 months, all at once.

> Those DAG Runs will all have been started on the same actual day, but their data intervals will cover those last 3 months, and that's what all the tasks, operators and sensors inside the DAG look at when they run.

Reviewer 2: ```suggestion
Those DAG Runs will all have been started on the same actual day, but each DAG run will have a data interval covering a single day in that 3 month period, and that data interval is all the tasks, operators and sensors inside the DAG look at when they run.
```

I think this is more clear, but I'm not hellbent on this one.

Author: Definitely clearer than the previous wording!

You can also say a task can only run if the *previous* run of the task in the previous DAG Run succeeded. To use this, you just need to set the ``depends_on_past`` argument on your Task to ``True``.

> Note that if you are running the DAG at the very start of its life---specifically, its first ever *automated* run---then the Task will still run, as there is no previous run to depend on.

Reviewer 2: ```suggestion
Note that if you are running the DAG at the very start of its life (specifically, its first ever *automated* run) then the Task will still run, as there is no previous run to depend on.
```

This looks better to my eyes?
